lib/addons/vpc-cni/index.ts

```diff
@@ -1,15 +1,185 @@
+  enablePodEni?: string;
+  /**
+  * `ENABLE_PREFIX_DELEGATION` Environment Variable. Format `boolean`.
+  * To enable prefix delegation on nitro instances. Setting 
+  * ENABLE_PREFIX_DELEGATION to true will start allocating a prefix (/28 for IPv4 and /80 for IPv6) instead of a secondary IP in the ENIs subnet. 
+  */
+  enablePrefixDelegation?: string;
+  /**
+  * `WARM_ENI_TARGET` Environment Variable. Format `integer`.
+  * Specifies the number of free elastic network interfaces (and all of their 
+  * available IP addresses) that the ipamd daemon should attempt to keep 
+  * available for pod assignment on the node. 
+  */
+  warmEniTarget?: string;
+  /**
+  * `WARM_PREFIX_TARGET` Environment Variable. Format `integer`.
+  * Specifies the number of free IPv4(/28) prefixes that the ipamd daemon 
+  * should attempt to keep available for pod assignment on the node.
+  */
+  warmPrefixTarget?: string;
+}
+
+/**
+ * Implementation of VpcCni EKS add-on with Advanced Configurations.
  */
 export class VpcCniAddOn extends CoreAddOn {
 
-    constructor(version?: string) {
-        super({
-            addOnName: "vpc-cni",
-            version: version ?? "v1.12.0-eksbuild.1",
-            saName: "vpc-cni"
-        });
+    
+  constructor(props?: VpcCniAddOnProps) {
+    super({
+      addOnName: "vpc-cni",
+      version: props?.version ?? "v1.12.1-eksbuild.2",
+      saName: "vpc-cni",
+      configurationValues: populateVpcCniConfigurationValues(props)  
+    });
+      
+  }
+}
+
+function populateVpcCniConfigurationValues(props?: VpcCniAddOnProps): Values {
+  if (props === null) {
+    return {};
+  }
+
+  let values: Values = {};
+  const result : Values = {
+    env: {
+        ADDITIONAL_ENI_TAGS: props?.additionalEniTags, 
+        ANNOTATE_POD_IP: props?.annotatePodIp,
+        AWS_VPC_CNI_NODE_PORT_SUPPORT: props?.awsVpcCniNodePortSupport,
+        AWS_VPC_ENI_MTU: props?.awsVpcEniMtu,
+        AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER: props?.awsVpcK8sCniConfigureRpfilter,
+        AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG: props?.awsVpcK8sCniCustomNetworkCfg,
+        AWS_VPC_K8S_CNI_EXTERNALSNAT: props?.awsVpcK8sCniExternalSnat,
+        AWS_VPC_K8S_CNI_LOGLEVEL: props?.awsVpcK8sCniLogLevel,
+        AWS_VPC_K8S_CNI_LOG_FILE: props?.awsVpcK8sCniLogFile,
+        AWS_VPC_K8S_CNI_RANDOMIZESNAT: props?.awsVpcK8sCniRandomizeSnat,
+        AWS_VPC_K8S_CNI_VETHPREFIX: props?.awsVpcK8sCniVethPrefix,
+        AWS_VPC_K8S_PLUGIN_LOG_FILE: props?.awsVpcK8sCniLogFile,
+        AWS_VPC_K8S_PLUGIN_LOG_LEVEL: props?.awsVpcK8sPluginLogLevel,
+        DISABLE_INTROSPECTION: props?.disableIntrospection,
+        DISABLE_METRICS: props?.disableMetrics,
+        DISABLE_NETWORK_RESOURCE_PROVISIONING: props?.disablenetworkResourceProvisioning,
+        ENABLE_POD_ENI: props?.enablePodEni,
+        ENABLE_PREFIX_DELEGATION: props?.enablePrefixDelegation,
+        WARM_ENI_TARGET: props?.warmEniTarget,
+        WARM_PREFIX_TARGET: props?.warmPrefixTarget
     }
+  }
```

Comment:
Please address github action warning missing semicolon here. Does lint catch it?

Reply:
Nope i didnt see this in the link. If you the PR passed all checks.

Reply:
I fixed it though.